projects/NodeBB/services/nodebb/module.nix

```
@@ -6,13 +6,260 @@
 }:
 let
   cfg = config.services.nodebb;
+  settingsFormat = pkgs.formats.json { };
+
+  # Similar to lib.mapAttrsRecursive but flattens the result
+  mapAttrsRecursive' =
+    f: set:
+    let
+      recurse =
+        acc: path:
+        lib.mapAttrsToList (
+          name: value:
+          let
+            path' = path ++ [ name ];
+          in
+          if lib.isAttrs value then recurse acc path' value else acc ++ [ (f path' value) ]
+        );
+    in
+    lib.listToAttrs (lib.flatten (recurse [ ] [ ] set));
+
+  configFile =
+    let
+      # nodebb will try redis if in config.json even if postgres is selected
+      excludeIfNot = value: if cfg.settings.database != value then [ value ] else [ ];
+      settings = lib.removeAttrs cfg.settings (
+        (excludeIfNot "mongo") ++ (excludeIfNot "postgres") ++ (excludeIfNot "redis")
+      );
+    in
+    settingsFormat.generate "config.json" (
+      # { a = { b = 0; }; } -> { "a:b" = 0; }
+      # https://github.com/NodeBB/NodeBB/blob/3e961257ec0904dbc3b3c64dab3d4cbdffcfbbd7/src/install.js#L177
+      settings // (mapAttrsRecursive' (path: lib.nameValuePair (lib.concatStringsSep ":" path)) settings)
```

We could use this but the `lib` here doesn't have it.

```suggestion
      settings // (lib.flattenAttrs ":" settings)
```